ehrapy/api/io/read.py

```diff
@@ -31,62 +25,69 @@ class DataReader:
 
     @staticmethod
     def read(
-        file_path: Union[Path, str],
-        download_file_name: Optional[str] = None,
+        dataset_path: Union[Path, str],
         extension: Optional[str] = None,
         delimiter: Optional[str] = None,
-        index_column: Union[str, Optional[int]] = None,
-        columns_obs_only: Optional[List[Union[str]]] = None,
+        index_column: Optional[Union[Dict[str, Union[str, int]], Union[str, int]]] = None,
```

Comment:
Stuff like this really shows that we need Python 3.10 haha

Reply:
;D and we need multi dispatch as well